.github/workflows/main.yml

```diff
@@ -7,31 +7,52 @@ jobs:
     name: ${{ matrix.platform.name }}
     runs-on: ${{ matrix.platform.os }}
     strategy:
+      fail-fast: false
       matrix:
         platform:
-        - { name: Windows, os: windows-latest }
-        - { name: Linux,   os: ubuntu-latest, flags: -GNinja }
+        - { name: Windows, os: windows-latest, msvc: true }
+        - { name: Linux,   os: ubuntu-latest }
         - { name: MacOS,   os: macos-latest }
     steps:
-    - name: Setup Linux dependencies
-      if: runner.os == 'Linux'
-      run: |
-        sudo apt-get update
-        sudo apt-get install cmake ninja-build libgl1-mesa-dev libglu1-mesa-dev
+    - uses: ilammy/msvc-dev-cmd@v1
+      if: ${{ matrix.platform.msvc }}
+      with:
+        arch: x64
     - name: Get sdl2-compat sources
       uses: actions/checkout@v3
-    - name: Get SDL3 headers
-      uses: actions/checkout@v3
+    - name: Install Ninja
+      if: ${{ !contains(matrix.platform.shell, 'msys2') }}
+      uses: turtlesec-no/get-ninja@main
+    - name: Set up SDL3
+      uses: libsdl-org/setup-sdl@main
       with:
-        repository: libsdl-org/SDL
-        ref: main
-        path: SDL3
+        cmake-generator: Ninja
+        version: 3-head
+        sdl-test: true
+        shell: ${{ matrix.platform.shell }}
+        add-to-environment: true
+    - name: Set up Linux dependencies
+      if: ${{ runner.os == 'Linux' }}
+      run: |
+        sudo apt-get update -y
+        sudo apt-get install -y libgl1-mesa-dev libglu1-mesa-dev
+    - name: Configure (CMake)
+      shell: sh
+      run: |
+        cmake -B build -GNinja
+    - name: Build (CMake)
+      run: |
+        cmake --build build/
+    - name: Run build-time tests (CMake)
+      if: ${{ false }} # FIXME: enable build-time tests on CI
```

Comment:
I left running the self tests disabled.
A follow-up PR will re-enable this line.